New module - na_ontap_ldap

zeten30 opened the pull request:
##### SUMMARY
New module -  na_ontap_ldap - [Enable/Disable LDAP on SVM/vserver](https://www.netapp.com/us/media/tr-4073.pdf)

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
na_ontap_ldap

zeten30 committed:
New module - na_ontap_ldap

zeten30 committed:
Fixing module documentation - ansible-test sanity

lonico commented on lib/ansible/modules/storage/netapp/na_ontap_ldap.py:
```diff
@@ -0,0 +1,226 @@
+author: Milan Zink (@zeten30) <zeten30@gmail.com>/<mzink@redhat.com>
+
+description:
+- Create, modify or delete LDAP on NetApp ONTAP SV/vserver
```
SVM/vserver

zeten30 replied:
Resolved by commit 178b47169348d646fabf11b4b5915249dad6d6f7

lonico commented on lib/ansible/modules/storage/netapp/na_ontap_ldap.py:
```diff
@@ -0,0 +1,226 @@
+    required: true
+    type: str
+
+  client_config:
```
I would prefer to say “name”.  So each resource has a name, independently of the ZAPI internals.

zeten30 replied:
@lonico Ok, no issues with changing 'client_config' -> 'name'. I'll do it.

zeten30 replied:
I've just pushed updated code. Please check it.

zeten30 committed:
Fulfil merge request changes from comments

- 'client_config' option renamed to 'name'
- docstring updated
- imports placed according to Ansible module best practices

lonico reviewed (approved):
shipit

dprts commented:
all lgtm
shipit

carchi8py commented on lib/ansible/modules/storage/netapp/na_ontap_ldap.py:
```diff
@@ -0,0 +1,226 @@
+
+    def apply(self):
+        '''Call create/modify/delete operations.'''
+        current = self.get_ldap()
```
Could you add at the top of Apply the following line. This will create an ems log event for users with auto support turned on.

netapp_utils.ems_log_event("na_ontap_ldap", self.server)

zeten30 replied:
Sure, resolved by commit 178b47169348d646fabf11b4b5915249dad6d6f7

zeten30 committed:
Fixed docstring typo + netapp_utils.ems_log_event added

lonico reviewed (approved):
shipit

zeten30 commented:
Close & reopen to get rid of 'stale_ci' as suggested by ansibot :)

thedoubl3j reviewed (approved):
lgtm, shipit. module coming under netapp certified umbrella.
